src/pip/_internal/vcs/__init__.py

```diff
@@ -200,6 +200,14 @@ def _is_local_repository(self, repo):
         drive, tail = os.path.splitdrive(repo)
         return repo.startswith(os.path.sep) or drive
 
+    def is_repository_directory(self, path):
```

Comment:
nit: couldn't this be a classmethod also ?

Reply:
Thanks for reviewing! Interesting observation / question. I suppose so, but I have a feeling that could be true for a lot of the other methods, too (e.g. if you look technically at what the method uses). My personal feeling is that we should expose something as a class method only when it’s primarily for use in the absence of an object. In this case it’s used equally for both.

Reply:
I will make this change, FYI.